### PR TITLE
feat: Observability for RayServe and vLLM GPU

### PR DESCRIPTION
**Pull request by shivam-dubey-1**
### What does this PR do?

This PR provides observability with Prom and Grafana for rayserve-vllm-gpu pattern

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### Motivation

#586 
#645 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [x] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: website/docs/gen-ai/inference/GPUs/vLLM-rayserve.md [hunk 1]
```diff
@@ -379,6 +379,111 @@ In summary, the theory of relativity is a groundbreaking theory in physics that
...
+```
+`Note:` kubectl port-forward is not recommended for production use. Refer to [this Grafana document](https://grafana.com/tutorials/run-grafana-behind-a-proxy/) for exposing Grafana behind a reverse proxy.
+
+**Import Open Source Grafana Dashboards**
+
+- Create new dashboard by importing JSON file via Dashboards menu.
+- Click 'Dashboards' icon in left panel, 'New', 'Import', then 'Upload JSON file'.
+- Choose a JSON file.
+   - `Case 1:` If you are using Ray 2.24.0, you can use the sample config files in [GitHub repository](https://github.com/awslabs/data-on-eks/tree/main/ai-ml/jark-stack/terraform/monitoring/ray-dashboards). The file names have a pattern of xxx_grafana_dashboard.json.
+   - `Case 2:` Otherwise, you should import the JSON files from `/tmp/ray/session_latest/metrics/grafana/dashboards/` in the head Pod. You can use `kubectl cp` to copy the files from the head Pod to your local machine.
```
[the remark this answers is not in the document]
Comment (shivam-dubey-1) [reply website/docs/gen-ai/inference/GPUs/vLLM-rayserve.md:474]: In this step, users can obtain JSON files that indicate the version of the Ray cluster they are using, for example:
rayClusterConfig:
  rayVersion: '2.xx.0'
Currently, we are providing the JSON files for version 2.24, but later, if users are in version 2.50, they can retrieve these JSONs from the pods.

File: ai-ml/jark-stack/terraform/addons.tf [hunk 2]
```diff
@@ -143,6 +143,31 @@ module "eks_blueprints_addons" {
...
+  #---------------------------------------
+  # Prommetheus and Grafana stack
+  #---------------------------------------
+  #---------------------------------------------------------------
+  # 1- Grafana port-forward `kubectl port-forward svc/kube-prometheus-stack-grafana 8080:80 -n kube-prometheus-stack`
+  # 2- Grafana Admin user: admin
+  # 3- Get sexret name from Terrafrom output: `terraform output grafana_secret_name`
+  # 3- Get admin user password: `aws secretsmanager get-secret-value --secret-id <REPLACE_WIRTH_SECRET_ID> --region $AWS_REGION --query "SecretString" --output text`
+  #---------------------------------------------------------------
+  enable_kube_prometheus_stack = true
```
[the remark this answers is not in the document]
Comment (shivam-dubey-1) [reply ai-ml/jark-stack/terraform/addons.tf:155]: Sure, I can add

[the remark this answers is not in the document]
Comment (shivam-dubey-1) [reply ai-ml/jark-stack/terraform/addons.tf:155]: We have the addon now

File: ai-ml/jark-stack/terraform/addons.tf [hunk 3]
```diff
@@ -143,6 +143,41 @@ module "eks_blueprints_addons" {
...
+    ],
+  }
+  
+  #---------------------------------------
+  # CloudWatch metrics for EKS
+  #---------------------------------------
+  enable_aws_cloudwatch_metrics = true
+  aws_cloudwatch_metrics = {
+    values = [templatefile("${path.module}/helm-values/aws-cloudwatch-metrics-values.yaml", {})]
+  }
```
Comment (vara-bonthu) [inline ai-ml/jark-stack/terraform/addons.tf:177]: We should switch to EKS managed add-on for CloudWatch and remove this. Please see the BioNemo PR as an example https://github.com/awslabs/data-on-eks/pull/641/files#diff-0222ef610acbd44e74c858a5505a877ad82439b934b0667e3f81dc91d3a247fdR26

Comment (vara-bonthu) [reply ai-ml/jark-stack/terraform/addons.tf:177]: You can do this in the second PR.

**Review by vara-bonthu — APPROVED** [review]
lgtm

**Comment by vara-bonthu** [comment]
Verify pre-commit Min TF check please

**Pull request merged**